### PR TITLE
Feat/371 client download

### PR DESCRIPTION
**Pull request by creativeyann17**
- [x] add config endpoint
- [x] update download links endpoint
- [x] filter user resources permissions
- [x] clean-up code
- [ ] TUs ?
- [x] add resources policy config (cqdg/clin...)

### CODE REVIEW

File: app/services/keycloak/PermsService.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  def checkPermissions(token: String, files: Array[String]): Set[String] = {
+    try {
+      // need to convert the user token into authorization token, because the user token is from a public client
+      // and authorization token is from the client with credentials where the resources are declared
+      val authorizationToken = authzClient.authorization(token).authorize().getToken
+      val perms = authzClient.protection().introspectRequestingPartyToken(authorizationToken);
+      val permsNames = perms.getPermissions.asScala.map(_.getResourceName).toSet
+      files.filter(permsNames.contains).toSet
+    } catch {
+      // if 403 user a no permissions at all, return empty set
```
Comment (celinepelletier) [inline app/services/keycloak/PermsService.scala]: if 403 user HAS no permissions at all ;)

File: app/controllers/HomeController.scala [hunk 2]
```diff
@@ -31,13 +33,17 @@ class HomeController @Inject()(val controllerComponents: ControllerComponents, s
...
-  def get(file: String): Action[AnyContent] = authAction { implicit request: UserRequest[AnyContent] =>
-    val url = s3.presignedUrl(bucket, prefix, file)
-    request.getQueryString("format") match {
-      case Some("json") =>
-        Ok(Json.toJson(Map("url" -> url.toString)))
-      case _ =>
-        Found(url.toString)
+  def downloadLinks(): Action[AnyContent] = authAction { implicit request: UserRequest[AnyContent] =>
+    val token = request.headers.get(HeaderNames.AUTHORIZATION).get.replace("Bearer ", "")
+    val requestedFiles = request.body.asText.get.split("\n")
```
Comment (celinepelletier) [inline app/controllers/HomeController.scala]: I would convert to Set here to make sure that there is no duplicates

File: app/services/keycloak/PermsService.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  val keycloakConfig = new org.keycloak.authorization.client.Configuration(
+    config.get[String]("auth.url"),
+    config.get[String]("auth.realm"),
+    config.get[String]("auth.client-id"),
+    Collections.singletonMap("secret",config.get[String]("auth.secret-key")), null
+  )
+  val authzClient: AuthzClient = AuthzClient.create(keycloakConfig);
+
+  def checkPermissions(token: String, files: Array[String]): Set[String] = {
```
Comment (celinepelletier) [inline app/services/keycloak/PermsService.scala]: a detail but I would either rename this function to retrieveAuthorizedFiles, or change the implementation to return in one shot 2 sets (one of authorized files, one of unauthorized files) : 

```

def checkPermission(token: String, files: Set[String]): (Set[String], Set[String]) = {
  // .....
  val permNames = Set("aaa", "ccc", "eee", "fff")
  
  val authorized = files intersect permNames
  val unauthorized = files diff permNames
  
  (authorized, unauthorized)
}

val files = Set("aaa", "bbb", "ccc", "ddd", "eee")
val (ok, nok) = checkPermission("token", files)
```

**Review by celinepelletier — APPROVED** [review]

File: app/services/keycloak/PermsService.scala (hunk 3 above)
Comment (creativeyann17) [reply app/services/keycloak/PermsService.scala]: nice that's cleaner indeed

**Review by celinepelletier — APPROVED** [review]

File: app/controllers/HomeController.scala [hunk 4]
```diff
@@ -31,13 +33,16 @@ class HomeController @Inject()(val controllerComponents: ControllerComponents, s
...
 
-  def get(file: String): Action[AnyContent] = authAction { implicit request: UserRequest[AnyContent] =>
-    val url = s3.presignedUrl(bucket, prefix, file)
-    request.getQueryString("format") match {
-      case Some("json") =>
-        Ok(Json.toJson(Map("url" -> url.toString)))
-      case _ =>
-        Found(url.toString)
+  def downloadLinks(): Action[AnyContent] = authAction { implicit request: UserRequest[AnyContent] =>
+    val token = request.headers.get(HeaderNames.AUTHORIZATION).get.replace("Bearer ", "")
```
Comment (jecos) [inline app/controllers/HomeController.scala]: I think you should get this information from UserRequest

**Review by jecos — CHANGES_REQUESTED** [review]

**Pull request merged**